### ISSUE TITLE
Instead of Capybara::Poltergeist::TimeoutError, Raise a relevant error when alert() / confirm() dialogs are opened

### ISSUE THREAD

**Issue by jimeh** [teampoltergeist/poltergeist#464]
### TLDR

If an `alert()` dialog is open when Poltergeist tries to tell PhantomJS to go to a URL, or otherwise perform an action that the alert dialog prohibits, please raise a relevant error and dismiss the dialog. Right now Poltergeist seems to just hang and eventually times out. After which all following tests timeout too. Selenium for example raises a `Selenium::WebDriver::Error::UnhandledAlertError` error and dismisses the dialog.

### Details

I recently spent a lot of time trying to track down a flickering issue with a set of Cucumber features. Nothing specifically failed, instead Capybara would start throwing a `Capybara::Poltergeist::TimeoutError` all of a sudden at random points. And once it had started happening, every single feature/scenario thereafter would also fail the same way.

The issue was a `alert()` dialog was on rare occasions opened from a JS error callback, which failed due to the database assertions of the scenario finishing before the JSON response made it back to the browser.

In Selenium however instead of every scenario thereafter failing, only the very next scenario would fail with a `Selenium::WebDriver::Error::UnhandledAlertError` exception, and the alert box was dismissed. That combined with the fact that I could see the alert for a split second in Firefox, helped me determine the actual cause.

In my case, all I had to do was wait for the JS success callback to fire before asserting that the database side that things had worked as they should.

**Comment by twalpole** [comment]
Behavior with current releases of Poltergeist is to by default close alerts, accept confirms, and return the default value for prompts - these default behaviors can be changed by using the capybara modal api each time a modal is expected.  I don't think it's possible for a page to get stuck with an open modal.  If this is still an issue for people with recent releases please provide a minimal test case and we can reopen
